player/src/main/java/com/focusit/jsflight/player/script/PlayerScriptProcessor.java

```diff
@@ -29,10 +28,14 @@
 {
     private static final Logger log = LoggerFactory.getLogger(PlayerScriptProcessor.class);
     private ScriptEngine engine;
+    private UserScenario scenario;
+
+    private Binding basicBinding;
```

Comment:
It seems like you've just added a concurrency issue. Just create basicBinding each time you need one